src/containers/OasisTrade.jsx

```diff
@@ -32,31 +42,45 @@ export class OasisTradeWrapper extends PureComponent {
         );
       }
     } else {
-      this.props.actions.setActiveTokenPair({
-        baseToken: params.baseToken, quoteToken: params.quoteToken
-      });
+      if(!this.props.activeTokenPair) {
+        this.props.actions.setActiveTokenPair({
+          baseToken: params.baseToken, quoteToken: params.quoteToken
+        });
+      }
       return null;
     }
   }
   render() {
+    const { tradedTokens, marketsData, defaultTimeSpan, loadingTradeHistory } = this.props;
     return this.redirect() || (
       <main>
-        <OasisMarketWidget/>
+        <OasisMarketWidget
+          tradedTokens={tradedTokens}
+          marketData={marketsData}
+          defaultTimeSpan={defaultTimeSpan}
+          loadingTradeHistory={loadingTradeHistory}
+        />
       </main>
     );
   }
 }
 
 export function mapStateToProps(state) {
   return {
-    validBaseTokensList: tokensSelectors.validBaseTokensList(state),
-    validQuoteTokensList: tokensSelectors.validQuoteTokensList(state)
+
+    validBaseTokensList: tokens.validBaseTokensList(state),
+    loadingTradeHistory: !trades.initialMarketHistoryLoaded(state),
+    validQuoteTokensList: tokens.validQuoteTokensList(state),
+    marketsData: trades.marketsData(state),
+    activeTokenPair: tokens.activeTokenPair(state),
+    tradedTokens: tokens.tradingPairs(state),
```

Comment:
`tradedTokens` value is taken from `tradingPairs` but then we have `tokenPair`. What's with the terms?`

Reply:
i unify those. done